### ISSUE TITLE
4.9.2.6 downloading miners again and again, mining never starts

### ISSUE THREAD

**Issue by bredita** [RainbowMiner/RainbowMiner#2793]
after upgrading to 4.9.2.6, rainbowminer keeps downloading the same miners over and over again.

Even after installing RBM in a new folder and going through setup in the console window, I have the same issue.

The console says:
WARNING: Download in progress: 40 miners left. Command windows might popup during extraction.
WARNING: Download in progress: 2 mining pre-requisites left.

The downloads appear in the downloads folder, but the log says:
[2024-06-02 11:23:54] WARNING: Downloader-error: Cannot find path 'D:\mining\RainbowMinerV4.9.2.6\Bin\gminer_2_94_windows64' because it does not exist.
[2024-06-02 11:23:54] WARNING: Cannot download D:\mining\RainbowMinerV4.9.2.6\Bin\GPU-Gminer294\miner.exe distributed at https://github.com/RainbowMiner/miner-binaries/releases/download/v2.94-gminer/gminer_2_94_windows64.zip. 
[2024-06-02 11:23:59] WARNING: Downloader-error: Cannot find path 'D:\mining\RainbowMinerV4.9.2.6\Bin\gminer_2_75_windows64' because it does not exist.
...

on every new round, RBM downloads the same miners again, and the zip appears in the download folder.

If I create the bin folder for a miner manually, it gets deleted when I start RBM.

If I try to download the files manually in my browser, they download just fine.

Mining never starts.

**Comment by bredita** [comment]
http://localhost:4000/downloadlist:


[
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v6.21.3.11-xmrig/xmrig-6.21.3.11-msvc-win64.7z",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\ANY-Xmrig\\xmrig.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v2.5.5-srbminermulti/SRBMiner-Multi-2-5-5-win64.zip",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\ANY-SRBMinerMulti\\SRBMiner-MULTI.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v2.25-teamblack/TeamBlackMiner_2_25_cuda_12_2.7z",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\GPU-Teamblack\\TBMiner.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v6.21.3.11-xmrig/xmrig-6.21.3.11-msvc-win64.7z",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\ANY-Xmrig\\xmrig-mo.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v1.17.3-rigel/rigel-1.17.3-win.zip",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\NVIDIA-Rigel\\rigel.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v21.1.5-bzminer/bzminer_v21.1.5_windows.zip",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\GPU-BzMiner\\bzminer.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v24.3-jayddee/cpuminer-opt-24.3-windows.zip",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\CPU-JayDDee\\cpuminer-avx512-sha-vaes.exe",
    "IsMiner": true
  },
  {
    "URI": "https://github.com/RainbowMiner/miner-binaries/releases/download/v6.21.0-xmrigcuda/xmrig-cuda-6.21.1-cuda12_4-win64.7z",
    "Path": "D:\\mining\\RainbowMiner\\Bin\\ANY-Xmrig\\xmrig-cuda.dll",
    "IsMiner": false
  }
]

**Comment by RainbowMiner** [comment]
Oh dear! Might be the new download strategy. I will check asap!

**Comment by RainbowMiner** [comment]
Just a short question: is the RainbowMiner path excluded from your antivirus? Sometimes miners get quarantained just after the download.

**Comment by bredita** [comment]
yes

**Comment by bredita** [comment]
auto update to 4.9.2.6  from 4.9.2.4 failed (on   "Timestamp": "5/30/2024 8:30:41 PM")

and I updated manually as described in the docs (unzipped new files into the folder).

but thats probably not relevant, as i also tried a new fresh install to another folder today, and it still has the same issue

**Comment by RainbowMiner** [comment]
Thank you for your reply! How did the auto-update fail? Was there any error message?

**Comment by bredita** [comment]
can't remember what the error message was exactly, sorry.

But it told me in the console that auto update had failed. So I did the manual update as described here:
https://github.com/RainbowMiner/RainbowMiner/blob/master/Doc/UPDATE.md

But again, I downloaded it again today, unzipped to a new folder below my mining folder (which is excluded from windows defender/AV) and that didn't work either. Then it wanted to download all miners again, and all seemed to fail.

Does it work for you if you just unzip rainbowminer to a new folder and create a new setup the normal way using install.bat?

**Comment by RainbowMiner** [comment]
I just tried the update path for an old instance - worked.
Now I will try a new install onto a drive other that C:

**Comment by bredita** [comment]
It works here too, mining is OK on the **old instance** I have always been using.
It just downloads those 6 miners on every new round, fails to install them, then repeats.

I noticed the problem in my network yesterday, noticed that my pc had been download some 150+ GB from github in the last 24 hours or so.

**Comment by RainbowMiner** [comment]
My new install on drive D: works too. All miners downloaded. So it's only these 6 miners hanging? I will try to remove them from one of the miners and see what happens.

**Comment by bredita** [comment]
Must be something at my end then. Will try another fresh install when i get home.

**Comment by bredita** [comment]
So, I've tried a clean install from the large zip file in releases, the small zip file, and I've also tried cloning the git repository into a folder.

I can't get any of them to work.

Have tried with windows AV disabled.

all settings default, with just a btc address added, 
selected:
no API user/pw, 
no nicehash, 
enable cpu mining

The log says:

`[2024-06-02 18:41:23] WARNING: Download in progress: 46 miners left. Command windows might popup during extraction.
[2024-06-02 18:41:23] WARNING: Download in progress: 2 mining pre-requisites left.
[2024-06-02 18:41:23] INFO: Start waiting 48 seconds before next run. 
[2024-06-02 18:41:25] WARNING: Downloader-error: Cannot find path 'C:\mining_test2\RainbowMiner_GIT\RainbowMiner\Bin\t-rex-0.19.14-win-cuda11.1' because it does not exist.
[2024-06-02 18:41:25] WARNING: Cannot download C:\mining_test2\RainbowMiner_GIT\RainbowMiner\Bin\NVIDIA-Trex19\t-rex.exe distributed at https://github.com/RainbowMiner/miner-binaries/releases/download/v0.19.14-trex/t-rex-0.19.14-win-cuda11.1.zip. 
[2024-06-02 18:41:28] WARNING: Downloader-error: Cannot find path 'C:\mining_test2\RainbowMiner_GIT\RainbowMiner\Bin\t-rex-0.26.8-win' because it does not exist.
[2024-06-02 18:41:28] WARNING: Cannot download C:\mining_test2\RainbowMiner_GIT\RainbowMiner\Bin\NVIDIA-Trex\t-rex.exe distributed at https://github.com/RainbowMiner/miner-binaries/releases/download/v0.26.8-trex/t-rex-0.26.8-win.zip. 

...

And yes, I am running it as admin

**Comment by RainbowMiner** [comment]
How superstrange! Can you send me a zip of your Logs folder by email to info@rbminer.net, please?

**Comment by RainbowMiner** [comment]
One idea: could you try to install the older release v4.9.2.4 (the small zip file) and try with that? If this install doesn't fail, we could at least rule out external problems.

**Comment by bredita** [comment]
Thank YOU for your help.

I have tried all these:
02.06.2024  20:54    <DIR>          RainbowMinerV4.8.8.2_win
02.06.2024  20:40    <DIR>          RainbowMinerV4.9.1.7_win
02.06.2024  20:35    <DIR>          RainbowMinerV4.9.1.9_win
02.06.2024  20:31    <DIR>          RainbowMinerV4.9.2.2_win
02.06.2024  20:26    <DIR>          RainbowMinerV4.9.2.3_win
02.06.2024  20:19    <DIR>          RainbowMinerV4.9.2.4_win

and they all do exactly the same. 
They say they want to download X miners.
They start downloading.
After a while there are X+1 files in the download folder (the extra file is the config file)
Then one file disappears, and a short while later it re-appears with a new timestamp.
This repeats until I kill it.

Must be something strange with my pc.

I have also tried on the two other user accounts I have on this pc, same problem there.

Have checked if any new apps have been installed on my pc. Nothing very exciting in the last few weeks, but a few windows/edge browser updates. There's one big update waiting. "cumulative update preview for Win 11 version 32H2". Installing that now. 

I understand if you want to give up. I do.

**Comment by RainbowMiner** [comment]
> and they all do exactly the same.

That is in fact good news. It must be some strange problem with your PC, then, because I can rule out the new download strategy.

Let's see - one thing you could try is:
open cmd window
cd to your RainbowMiner folder
enter `.\7z.exe`
Does it start the 7z - and what is the version?
It should be `7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21` written at the beginning

**Comment by RainbowMiner** [comment]
> Have tried with windows AV disabled.

Don't disable it - add the RainbowMiner installation path to the exceptions of Defender.

**Comment by bredita** [comment]
> > and they all do exactly the same.
> 
> That is in fact good news. It must be some strange problem with your PC, then, because I can rule out the new download strategy.
> 
> Let's see - one thing you could try is: open cmd window cd to your RainbowMiner folder enter `7z.exe` Does it start the 7z - and what is the version? It should be `7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21` written at the beginning

thats the one.

C:\mining_test2\RainbowMiner_GIT\RainbowMiner>7z.exe

7-Zip 19.00 (x64) : Copyright (c) 1999-2018 Igor Pavlov : 2019-02-21

**Comment by whitindllc** [comment]
have you disabled built in av for the time being?

**Comment by bredita** [comment]
it's obviously able to download the miners. How can I check if it does the right thing after having downloaded it?

what is supposed to happen?

- should it download one miner, unzip it (to where?), install it (to where?) and then move on to install the next miner?
or
- should it download all miners, then unzip and install them?

**Comment by bredita** [comment]
I may have fixed it for you:

Modules/Include.psm1, line 2782, changed from:
```
        } else {
            $Params = @{
                FilePath     = "7z.exe"
                ArgumentList = "x `"$FromFullPath`" -o`"$ToFullPath`" -y -spe"
                WindowStyle  = "Hidden"
            }
        }
```

to:

```
        } else {
	Write-Log -Level Info "we're NOT running on linux...."
            $Params = @{
                FilePath     = ".\7z.exe"
                ArgumentList = "x `"$FromFullPath`" -o`"$ToFullPath`" -y -spe"
                WindowStyle  = "Hidden"
            }
        }
```
And now it downloads miners, unzips them to the bin folder, and does whatever it is supposed to do after that.

(Ignore that Write-Log line, I added lots of debug-logging to pinpoint the issue)

please don't ask me why it works with ./ and not without ./ before 7z.exe

**Comment by whitindllc** [comment]
are you running start,exe as admin? i have that same parameter and don't have that issue at all

![image](https://github.com/RainbowMiner/RainbowMiner/assets/128111946/edc4b46c-d7b0-4f1d-a673-61756ca91e09)

**Comment by bredita** [comment]
Yes, I'm running it as admin, ofcourse.

I found out what caused the issue.

I installed LUA in order to make that script i mentioned in issue "update powerprice via script? #2769", the script that currently calculates my powerprice every 5 mins and switches on my asic miners via some "home automation" switches when it's profitable to do so.

This LUA install put an old version of 7z in my path:
```

C:\Windows\System32>where 7z
C:\Program Files (x86)\Lua\5.1\7z.exe

C:\Windows\System32>7z
7-Zip 9.10 beta  Copyright (c) 1999-2009 Igor Pavlov  2009-12-22
```

This version of 7z does not work with RainbowMiner, I guess.

After forcing rainbowminer to use its own 7z.exe (by adding .\ first in the FilePath parameter), everything works again.

All miners have now been downloaded in my 4.9.2.6 instance, using all my old settings.
RainbowMiner is finally mining again on my "Gaming PC" :) 

One question: What's the difference between profit and earnings?
Total profit/earnings:
 0.536/0.684 USD/Day
